src/components/Api.js

```diff
@@ -0,0 +1,144 @@
+export default class Api {
+  constructor(options) {
+    this._url = options.url;
+    this.authorization = options.authorization;
+    this._headers = {
+      authorization: this.authorization,
+      "Content-Type": "application/json",
+    };
+  }
+
+  _processResponse = (res) => {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
+  };
+
+  getCurrentUser(endPoint) {
+    return fetch(`${this._url}${endPoint}`, {
+      method: "GET",
+      headers: this._headers,
+    }).then(this._processResponse);
+  }
+
+  editCurrentUser(endPoint, input) {
```

Comment:
Needs correction: please remove `endPoint` argument from all the api methods, since each of them is responsible for the specific patch, it can't be changed

Reply:
I'm assuming that means I would type the endpoint string into each of my API methods, but wouldn't that make the code less reusable?  

I made the changes but was just hoping for some clarification on the pros and cons of doing it both ways :D

Reply:
It won't make your code reusable, since each api method should be responsible for only one endpoint. If you will pass the endpoint via arguments there will be more chances to make mistakes and there will be more sources of truth.